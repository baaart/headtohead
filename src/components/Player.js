import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd'
import styled from 'styled-components';

import Stats from './Stats'
import CardHeader from './CardHeader';

const PlayerWrapper = styled.div`
  margin: 10px;
` 
const CardContent = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: start;
`
const Avatar = styled.img`
  width: 50%;
  height: 50%;
`

function Player({...props}) {
  return (
    <PlayerWrapper>
      <Card 
        title={<CardHeader title={`${props.data.firstname} ${props.data.lastname}`} 
                           url={props.data.country.picture} />}
      >
        <CardContent>
          <Avatar src={props.data.picture} />
          <Stats data={props.data.data} />
        </CardContent>
      </Card>
    </PlayerWrapper>
  );
}

Player.propTypes = {
  data: PropTypes.object.isRequired
};

export default Player;
