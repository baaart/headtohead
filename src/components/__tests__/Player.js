import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Player from '../Player';
import { fakeData } from '../__mocks__/fakeData'

describe('Player', () => {
  it('Matches snapshot', () => {
    const wrapped = shallow(<Player data={fakeData.players[0]} />);
    expect(wrapped).toMatchSnapshot();
  })
});
