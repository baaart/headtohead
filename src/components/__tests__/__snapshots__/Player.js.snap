// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    data={
      Object {
        "country": Object {
          "code": "SUI",
          "picture": "https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png",
        },
        "data": Object {
          "age": 33,
          "height": 183,
          "last": Array [
            1,
            1,
            1,
            0,
            1,
          ],
          "points": 1784,
          "rank": 21,
          "weight": 81000,
        },
        "firstname": "Stan",
        "lastname": "Wawrinka",
        "picture": "https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg",
        "sex": "M",
        "shortname": "S.WAW",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Card
        title={
          <CardHeader
            title="Stan Wawrinka"
            url="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
          />
        }
      >
        <ForwardRef>
          <ForwardRef
            src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
          />
          <Stats
            data={
              Object {
                "age": 33,
                "height": 183,
                "last": Array [
                  1,
                  1,
                  1,
                  0,
                  1,
                ],
                "points": 1784,
                "rank": 21,
                "weight": 81000,
              }
            }
          />
        </ForwardRef>
      </Card>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef
            src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
          />
          <Stats
            data={
              Object {
                "age": 33,
                "height": 183,
                "last": Array [
                  1,
                  1,
                  1,
                  0,
                  1,
                ],
                "points": 1784,
                "rank": 21,
                "weight": 81000,
              }
            }
          />
        </ForwardRef>,
        "title": <CardHeader
          title="Stan Wawrinka"
          url="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
            />,
            <Stats
              data={
                Object {
                  "age": 33,
                  "height": 183,
                  "last": Array [
                    1,
                    1,
                    1,
                    0,
                    1,
                  ],
                  "points": 1784,
                  "rank": 21,
                  "weight": 81000,
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "src": "https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-htoDjs",
                "isStatic": true,
                "rules": Array [
                  "
  width: 50%;
  height: 50%;
",
                ],
              },
              "displayName": "styled.img",
              "render": [Function],
              "styledComponentId": "sc-htoDjs",
              "target": "img",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "age": 33,
                "height": 183,
                "last": Array [
                  1,
                  1,
                  1,
                  0,
                  1,
                ],
                "points": 1784,
                "rank": 21,
                "weight": 81000,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  justify-content: start;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": true,
        "rules": Array [
          "
  margin: 10px;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Card
          title={
            <CardHeader
              title="Stan Wawrinka"
              url="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
            />
          }
        >
          <ForwardRef>
            <ForwardRef
              src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
            />
            <Stats
              data={
                Object {
                  "age": 33,
                  "height": 183,
                  "last": Array [
                    1,
                    1,
                    1,
                    0,
                    1,
                  ],
                  "points": 1784,
                  "rank": 21,
                  "weight": 81000,
                }
              }
            />
          </ForwardRef>
        </Card>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ForwardRef>
            <ForwardRef
              src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
            />
            <Stats
              data={
                Object {
                  "age": 33,
                  "height": 183,
                  "last": Array [
                    1,
                    1,
                    1,
                    0,
                    1,
                  ],
                  "points": 1784,
                  "rank": 21,
                  "weight": 81000,
                }
              }
            />
          </ForwardRef>,
          "title": <CardHeader
            title="Stan Wawrinka"
            url="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                src="https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg"
              />,
              <Stats
                data={
                  Object {
                    "age": 33,
                    "height": 183,
                    "last": Array [
                      1,
                      1,
                      1,
                      0,
                      1,
                    ],
                    "points": 1784,
                    "rank": 21,
                    "weight": 81000,
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "src": "https://i.eurosport.com/_iss_/person/pp_clubteam/large/325225.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": undefined,
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": true,
                  "rules": Array [
                    "
  width: 50%;
  height: 50%;
",
                  ],
                },
                "displayName": "styled.img",
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "img",
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "age": 33,
                  "height": 183,
                  "last": Array [
                    1,
                    1,
                    1,
                    0,
                    1,
                  ],
                  "points": 1784,
                  "rank": 21,
                  "weight": 81000,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-gzVnrw",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  justify-content: start;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-gzVnrw",
            "target": "div",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
  margin: 10px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
